### PR TITLE
Don't stop identifying non-final parameters once a final one is found

### LINKED ISSUE #2700: `prefer_final_parameters` false negative if the first parameter is final
**Describe the issue**
`prefer_final_parameters` gives false negative if the first parameter is `final`, but the rest of the parameters are not `final` and not reassigned.

**To Reproduce**
All of these should LINT

```dart
void badMixed2(final String label1, String label2) { // LINT
  print(label1);
  print(label2);
}

void badMixed3(final String label1, String label2, String label3) { // LINT
  print(label1);
  print(label2);
  print(label3);
}
```

**Expected behavior**
`prefer_final_parameters` should report `badMixed2` and `badMixed3`. However, `dart test -N prefer_final_parameters` fails if both of them are added into `linter/test_data/rules/prefer_final_parameters.dart`.

**Additional context**
Pinging the author of this rule @parlough

### PR DESCRIPTION
**Pull request by parlough**
This rule stopped looking for non-final parameters if it found a `final` one causing lints to not be triggered when they should have been. This modification only skips that parameter instead of stopping processing completely.

Fixes #2700

### CODE REVIEW

File: test_data/rules/prefer_final_parameters.dart [hunk 1]
```diff
@@ -68,11 +68,23 @@ void useGoodTypedClosureArgument() {
...
   print(good);
 }
 
+void badMixedFirst(final String goodFirst, String badSecond) { // LINT
+  print(goodFirst);
+  print(badSecond);
+}
+
+// LINT [+1]
+void badMixedMiddle(String goodFirst, final String badSecond, String badThird) { // LINT
```
Comment (parlough) [inline test_data/rules/prefer_final_parameters.dart:82]: @pq Is there a better way to identify multiple lints triggered on one line?

**Review by bwilkerson — APPROVED** [review]

File: test_data/rules/prefer_final_parameters.dart (hunk 1 above)
Comment (pq) [reply test_data/rules/prefer_final_parameters.dart]: Gross right? Not really, but at least it's possible...  Cheers to @a14n for that!

**Review by pq — COMMENTED** [review]
🎉

File: test_data/rules/prefer_final_parameters.dart (hunk 1 above)
Comment (a14n) [reply test_data/rules/prefer_final_parameters.dart]: You can be even more precise with something like:
```dart
// LINT [+2,column:length]
// LINT [+1,column:length]
void badMixedMiddle(String goodFirst, final String badSecond, String badThird) {
```

**Pull request merged**